src/components/app/details/AppHeader.tsx

```diff
@@ -0,0 +1,239 @@
+    selectedFilterTab,
+    setSelectedFilterTab,
+    groupFilterOptions,
+    selectedGroupFilter,
+    setSelectedGroupFilter,
+    openCreateGroup,
+    openDeleteGroup,
+    isSuperAdmin,
+}: AppHeaderType) {
+    const { appId } = useParams<{ appId }>()
+    const match = useRouteMatch()
+    const history = useHistory()
+    const location = useLocation()
+    const currentPathname = useRef('')
+    const [isMenuOpen, setMenuOpen] = useState(false)
+
+    const contextValue = useMemo(
+        () => ({
+            appListOptions,
+            isMenuOpen,
+            setMenuOpen,
+            selectedAppList,
+            setSelectedAppList,
+            selectedFilterTab,
+            setSelectedFilterTab,
+            groupFilterOptions,
+            selectedGroupFilter,
+            setSelectedGroupFilter,
+            openCreateGroup,
+            openDeleteGroup,
+            isSuperAdmin,
+            filterParentType: FilterParentType.app
+        }),
+        [
+            appListOptions,
+            isMenuOpen,
+            selectedAppList,
+            selectedFilterTab,
+            groupFilterOptions,
+            selectedGroupFilter,
+            isSuperAdmin,
+        ],
+    )
+
+    function onClickTabPreventDefault(event: React.MouseEvent<Element, MouseEvent>, className: string) {
+        const linkDisabled = (event.target as Element)?.classList.contains(className)
+        if (linkDisabled) {
+            event.preventDefault()
+        }
+    }
+
+    function handleEventClick(event) {
+        trackByGAEvent('App', event.currentTarget.dataset.action)
+        onClickTabPreventDefault(event, 'active')
+    }
+
+    useEffect(() => {
+        currentPathname.current = location.pathname
+    }, [location.pathname])
+
+    const handleAppChange = useCallback(
+        ({ label, value }) => {
+            const tab = currentPathname.current.replace(match.url, '').split('/')[1]
+            const newUrl = generatePath(match.path, { appId: value })
+            history.push(`${newUrl}/${tab}`)
+            ReactGA.event({
+                category: 'App Selector',
+                action: 'App Selection Changed',
+                label: tab,
+            })
+        },
+        [location.pathname],
+    )
+
+    const { breadcrumbs } = useBreadcrumb(
+        {
+            alias: {
+                ':appId(\\d+)': {
+                    component: <AppSelector onChange={handleAppChange} appId={appId} appName={appName} />,
+                    linked: false,
+                },
+                app: {
+                    component: <span className="cb-5 fs-16 dc__capitalize">devtron apps</span>,
+                    linked: true,
+                },
+            },
+        },
+        [appId, appName],
+    )
+
+    const renderAppDetailsTabs = () => {
+        return (
+            <ul role="tablist" className="tab-list">
+                <li className="tab-list__tab dc__ellipsis-right">
+                    <NavLink
```

Comment:
Can we keep all the tabs names in the array and iterate over that instead

Reply:
That is existing code so not going to do any major change there as this is urgent requirement. can be picked in next refactor.